crates/turborepo-lib/src/cli/mod.rs

```diff
@@ -75,6 +75,35 @@ impl Display for OutputLogsMode {
     }
 }
 
+/// Which binary was used for the current invocation
+pub enum BinaryType {
+    /// Local binary called directly
+    Local,
+    /// Local binary was called via a global binary
+    LocalDefer,
+    /// Global binary called directly (No local binary to handoff to)
+    Global,
+    /// Global binary called because repo inference failed
+    GlobalFallback,
+    /// Repo inference was explicitly skipped via `--skip-infer`` arg
+    SkipInfer,
+    /// Repo inference was explicitly skipped via `$TURBO_BINARY_PATH` env var
+    BinPath,
+}
```

Comment:
Few questions:

- Global just means anything that's not locally deferred right? E.g. `/path/to/turbo` is global the same as `npm i -g turbo && turbo`?

- What would be the binary type of `npx turbo` and is it possible to track that separately?

- I wonder if we could model this as two fields: `bin_location: local|global|explicit` and `bin_inference: defer|failed|skipped`?  It's not quite the same data (e.g. `BinPath` translates to `bin_type:explicit, bin_inference:skipped`, which can be confusing since it's not using `--skip-infer`)

Reply:
Good questions on the first two, I can test it out. 

And yea I had originally modeled this as a tuple much how you described it. But since the cross product of both location and inference aren't always valid (although I like the explicit) I just made it a simple enum array.

Reply:
sounds good to me! I think the `npx|pnpx|yarn dlx` case would especially be useful to know about, but it's not a blocker. We can merge this and see which pieces of data we want to dig deeper into